056-zk-kraft-migration.md

```diff
@@ -0,0 +1,254 @@
+
+```shell
+# Disable the migration
+# zookeeper.metadata.migration.enable=true
+
+# Remove ZooKeeper client configuration
+# zookeeper.connect=<connection-to-zookeeper>
+```
+
+Finally, the operator rolls the controllers and wait for them to be ready again.
+They are now disconnected from the ZooKeeper ensemble which is totally out of the picture.
+The FSM moves to the `FullKRaft` state and ZooKeeper is not used anymore.
+
+Before completing the migration, the `DualWrite` state is also the last opportunity for the user to rollback the migration process by applying the `strimzi.io/kraft-migration: pre-migration` on the `Kafka` custom resource.
+In this case, the operator configures the brokers with no ZooKeeper migration flag, and no KRaft controllers quorum connection information.
+Finally, the operator rolls the brokers and wait for them to be ready again.
+The FSM moves back to the `WaitBrokersInMigration`, so actually at the beginning of the Phase 2.
+
+#### Phase 5: Removing ZooKeeper
+
+At this point the ZooKeeper ensemble is out of the picture, still running but not used anymore.
+In the `FullKRaft` state, the operator expects the user to finalize the migration by removing ZooKeeper.
+The user removes the `spec.zookeeper` on the `Kafka` custom resource so that the operator can delete the ZooKeeper nodes.
+The FSM moves back to a `NoMigration` state (or null) and the operator upgrades `Kafka.status.migrationState` field accordingly.
+
+### Metrics
+
+During ZooKeeper to KRaft migration, there are some metrics that could be useful to check the status in each of the phases.
+More details in the [“Metrics”](https://cwiki.apache.org/confluence/display/KAFKA/KIP-866+ZooKeeper+to+KRaft+Migration#KIP866ZooKeepertoKRaftMigration-Metrics) section of KIP-866.
+
+The most interesting and useful ones are:
+
+| MBean                                                             | Description                                                                                                                        |
+|-------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------|
+| kafka.controller:type=KafkaController,name=MetadataType           | An enumeration of: ZooKeeper (1), KRaft (2), or Dual (3). The active controller reports this.                                      |
+| kafka.controller:type=KafkaController,name=MigratingZkBrokerCount | A count of ZK brokers that are registered with KRaft and ready for migration. This will only be reported by the active controller. |
+| kafka.controller:type=KafkaController,name=ZkMigrationState       | An enumeration of the possible migration states the cluster can be in. This is only reported by the active controller.             |
+
+They could be used the following way:
```

Comment:
Proposals are always easier to understand if you take a stance. Either make the case for actually using these things, or mention them as a rejected alternative, or don't mention them at all.